public/app/features/scenes/variables/interpolation/ScopedVarsVariable.ts

```diff
@@ -0,0 +1,169 @@
+   * Playing the role of a variable so the formatting logic does not need special handling for scopedVars
+   *
+   * The reason we are implementing the SceneVariable interface here instead of extending SceneObjectBase is because of circular dependencies.
+   * SceneObjectBase depends on sceneInterpolate which depends on this class.
+   **/
+
+  public validateAndUpdate?(): Observable<ValidateAndUpdateResult> {
+    throw new Error('Method not implemented.');
+  }
+
+  public isActive = false;
+  public parent?: SceneObject<SceneObjectState> | undefined;
+  public variableDependency?: SceneVariableDependencyConfigLike | undefined;
+
+  public subscribeToState(observer?: Partial<Observer<ScopedVarsProxyVariableState>> | undefined): Subscription {
+    throw new Error('Method not implemented.');
+  }
+
+  public subscribeToEvent<T extends BusEvent>(
+    typeFilter: BusEventType<T>,
+    handler: BusEventHandler<T>
+  ): Unsubscribable {
+    throw new Error('Method not implemented.');
+  }
+
+  public publishEvent(event: BusEvent, bubble?: boolean | undefined): void {
+    throw new Error('Method not implemented.');
+  }
+
+  public useState(): ScopedVarsProxyVariableState {
+    throw new Error('Method not implemented.');
+  }
+
+  public activate(): void {
+    throw new Error('Method not implemented.');
+  }
+
+  public deactivate(): void {
+    throw new Error('Method not implemented.');
+  }
+
+  public getSceneEditor(): SceneEditor {
+    throw new Error('Method not implemented.');
+  }
+
+  public getRoot(): SceneObject<SceneObjectState> {
+    throw new Error('Method not implemented.');
+  }
+
+  public getData(): SceneObject<SceneDataState> {
+    throw new Error('Method not implemented.');
+  }
+
+  public getVariables(): SceneVariables | undefined {
+    throw new Error('Method not implemented.');
+  }
+
+  public getTimeRange(): SceneTimeRange {
+    throw new Error('Method not implemented.');
+  }
+
+  public clone(state?: Partial<ScopedVarsProxyVariableState>): this {
+    throw new Error('Method not implemented.');
+  }
+
+  public Component(props: SceneComponentProps<SceneObject<ScopedVarsProxyVariableState>>): ReactElement {
+    throw new Error('Method not implemented.');
+  }
+
+  public Editor(props: SceneComponentProps<SceneObject<ScopedVarsProxyVariableState>>): ReactElement {
+    throw new Error('Method not implemented.');
+  }
+}
+
+interface FieldAccessorCache {
+  [key: string]: (obj: unknown) => unknown;
+}
+
+let scopedVarsVariable: ScopedVarsVariable | undefined;
+
+/**
+ * Reuses a single instance to avoid unnecessary memory allocations
+ */
+export function getSceneVariableForScopedVar(name: string, value: ScopedVar) {
+  if (!scopedVarsVariable) {
+    scopedVarsVariable = new ScopedVarsVariable({ name, value });
+  } else {
+    scopedVarsVariable.setState({ name, value });
+  }
+
+  return scopedVarsVariable;
+}
```

Comment:
Since javascript is single threaded it should not be an issue. We could change this to always create a new instance here instead of reusing the same instance, the setState as it written now will always publish a new event (and create a new even instance), so there is no real memory gain by reusing the same scene object